### PR TITLE
Plans 2023: Partial list of TODOs for NPM migration

### PR DESCRIPTION
**Pull request by chriskmnds**
Related to #

## Proposed Changes

Code TODOs. Not for merging.

## Testing Instructions

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/hooks/use-highlight-adjacency-matrix.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ interface HighlightAdjacencyMatrix {
 	};
 }
 
+// @todo npm-migration
+// pass currentPlanSlug as prop
+
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/hooks/use-highlight-adjacency-matrix.ts:17]: Handled in https://github.com/Automattic/wp-calypso/pull/76486

File: client/my-sites/plan-features-2023-grid/hooks/use-highlight-label.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ const useHighlightLabel = ( planName: string ) => {
 	const translate = useTranslate();
 	const selectedSiteId = useSelector( getSelectedSiteId );
 	const currentPlan = useSelector( ( state ) => getCurrentPlan( state, selectedSiteId ) );
+	// @todo npm-migration
+	// pass currentPlanSlug as prop instead
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/hooks/use-highlight-label.ts:13]: Handled in https://github.com/Automattic/wp-calypso/pull/76486

File: client/my-sites/plan-features-2023-grid/index.tsx [hunk 3]
```diff
@@ -984,13 +993,19 @@ const ConnectedPlanFeatures2023Grid = connect(
 				( planConstantObj.get2023PricingGridSignupStorageOptions &&
 					planConstantObj.get2023PricingGridSignupStorageOptions() ) ||
 				[];
+			// @todo npm-migration
+			// pass usePlanAvailability( siteId, plan-slug ) as prop
 			const availableForPurchase = isInSignup || isPlanAvailableForPurchase( state, siteId, plan );
+			// @todo npm-migration
+			// pass currentSitePlanSlug as prop (and update to compare with plan-slug))
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/index.tsx:1000]: Handled in https://github.com/Automattic/wp-calypso/pull/76486

File: client/my-sites/plan-features-2023-grid/index.tsx [hunk 4]
```diff
@@ -1014,6 +1029,12 @@ const ConnectedPlanFeatures2023Grid = connect(
 			};
 		} );
 
+		// @todo npm-migration
+		// pass manageHref as prop
+		// ^ removes a couple of dependencies
+		// pass selectedSiteSlug as prop
+		// pass currentSitePlanSlug as prop (if different?)
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/index.tsx:1036]: > pass currentSitePlanSlug as prop

Handled in https://github.com/Automattic/wp-calypso/pull/76486

File: client/my-sites/plans/hooks/use-is-large-currency.ts [hunk 5]
```diff
@@ -6,6 +6,10 @@ import type { PlanSlug } from '@automattic/calypso-products';
 
 const LARGE_CURRENCY_CHAR_THRESHOLD = 5;
 
+// @todo npm-migration
+// move under grid 2023
+// pass sideId as prop
+
```
Comment (chriskmnds) [inline client/my-sites/plans/hooks/use-is-large-currency.ts:12]: Both of these handled in https://github.com/Automattic/wp-calypso/pull/76491

File: client/my-sites/plan-features-2023-grid/components/actions.tsx [hunk 6]
```diff
@@ -60,6 +60,11 @@ const DummyDisabledButton = styled.div`
 	text-align: center;
 `;
 
+// @todo npm-migration
+// getPlanBillPeriod from plans data store
+// pass domainFromHomeUpsellFlow as prop
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/components/actions.tsx:65]: > pass domainFromHomeUpsellFlow as prop

I'm handling this one next. 
leads to some more general refactors in how we can pass and execute external actions

**Comment by aneeshd16** [comment]
I believe this should be removed as well:
https://github.com/Automattic/wp-calypso/blob/4910db94dde21aced9111956d87aae3389df408d/client/my-sites/plan-features-2023-grid/index.tsx#L650-L652

The plans grid by itself should not be redirecting users anywhere. It should defer action handling to its parent component. As a result, we'll also have to ensure that the `onUpgradeClick` props is always passed to this component.

File: client/my-sites/plan-features-2023-grid/index.tsx [hunk 7]
```diff
@@ -88,6 +88,10 @@ import type { IAppState } from 'calypso/state/types';
 
 import './style.scss';
 
+// @todo npm-migration
+// migrate *Logo to @automattic/components or behind a hook in a "logos" (or some other) data store
+// ^ preferred, but can also pass as props
```
Comment (ddc22) [inline client/my-sites/plan-features-2023-grid/index.tsx:93]: Migrating logos to components seem sensible. --> #76853

**Comment by chriskmnds** [comment]
Closing this in favor of all the [Plans: NPM Essential](https://github.com/Automattic/wp-calypso/issues?q=label%3A%22Plans%3A+NPM+essential%22+) issues created

**Pull request closed without merging**